machine/v1/types_controlplanemachineset.go

```diff
@@ -284,6 +289,35 @@ type GCPFailureDomain struct {
 	Zone string `json:"zone"`
 }
 
+// OpenStackFailureDomain configures failure domain information for the OpenStack platform.
+// +kubebuilder:validation:MinProperties:=1
+type OpenStackFailureDomain struct {
+	// availabilityZone is the nova availability zone in which the OpenStack machine provider will create the VM.
+	// If not specified, the VM will be created in the default availability zone specified in the nova configuration.
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
+	// rootVolume contains settings that will be used by the OpenStack machine provider to create the root volume attached to the VM.
+	// If not specified, no root volume will be created.
+	// +optional
+	RootVolume *RootVolume `json:"rootVolume,omitempty"`
+}
+
+// RootVolume represents the volume metadata to boot from.
+// The original RootVolume struct is defined in the v1alpha1 but it's not best practice to use it directly here so we define a new one
+// that should stay in sync with the original one.
+// +kubebuilder:validation:MinProperties:=1
```

Comment:
`kubebuilder:validation:MinProperties:=1` will force the `AvailabilityZone` field to be always set when specified. Which is equivalent to `+required`. You probably want to remove the validation line.

Reply:
but since `RootVolume` is optional, isn't it fine that we require at least one field in `RootVolume`?

Reply:
I just realized that there is a testsuite for CPMS (e.g. machine/v1/stable.controlplanemachineset.gcp.testsuite.yaml), I'll add one for OpenStack.

Sorry for that.

Reply:
Here I also added a case to the testsuite to make sure it works that way, when a rootVolume is specified, it can't be empty.